File: src/omnia/data/structure/observable/GenericMutationEvent.java
package omnia.data.structure.observable;

import omnia.data.structure.Collection;

class GenericMutationEvent<StateType, OperationsType extends Collection<?>>
    implements ObservableDataStructure.MutationEvent {
  private final StateType state;
  private final OperationsType operations;

  protected GenericMutationEvent(StateType state, OperationsType operations) {
    this.state = state;
    this.operations = operations;
  }

  @Override
  public StateType state() {
    return state;
  }

  @Override
  public OperationsType operations() {
    return operations;
  }
}

File: omnia/src/commonMain/omnia/data/structure/observable/writable/GenericMutationEvent.kt
package omnia.data.structure.observable.writable

import omnia.data.structure.Collection
import omnia.data.structure.observable.ObservableDataStructure

internal open class GenericMutationEvent<StateType : Any, OperationsType : Collection<*>> protected constructor(
  override val state: StateType,
  override val operations: OperationsType
) : ObservableDataStructure.MutationEvent